### ISSUE TITLE
Presence indicator in the profile element stuck at away status

### ISSUE THREAD

**Issue by Gummikavalier** [RocketChat/Rocket.Chat#33789]
### Description:

Presence indicator in user profile element / menu gets stuck in `Away` status after idling for the predefined time.

This does not concern actual status visible to other users.

This happens very often but not always. I haven't been able to pinpoint exact way to reproduce reliably.

### Steps to reproduce:

1. Set the status to `Online` from the top left profile menu.
2. Do not use the browser; wait that idle timeout turns the user status from `Online` to `Away` status
3. Focus back on the browser and write couple of messages to any channel

### Expected behavior:

`Away` status should change back to `Online`.

### Actual behavior:

When you focus to RC browser window, status may flash very briefly (tens of milliseconds) into green but after that remains orange at `Away` status.

(The user's presence status is correct when checked by other users. The issue is purely local.)

### Server Setup Information:

- Version of Rocket.Chat Server: 6.13.0
- License Type: Enterprise
- Number of Users: 500+
- Operating System: RHEL8
- Deployment Method: tar
- Number of Running Instances: 4
- DB Replicaset Oplog: Yes
- NodeJS Version: 14.21.3
- MongoDB Version: 6.0

### Client Setup Information

- Desktop App or Browser Version: At least the latest Firefox and RC desktop client. In both Windows and Linux.

### Additional context

Happens in both old and new preview UX.

Also happens in RC 7.0.0 rc.3.

**Comment by reetp** [comment]
> Also happens in RC 7.0.0 rc.3.

Is it in v7.0.0 ?

If so I'll refer it to the team (though you have EE so have you contacted them internally?)

**Comment by Gummikavalier** [comment]
Yes, I checked now on our test system (RC 7.0.0) and I'm still seeing this too.

Yup, we do have EE but I quite often prefer GitHub because it may trigger answers from other RC maintainers too, which has been helpful in some obscure issues through the years.

**Comment by Gummikavalier** [comment]
I made a custom CSS change, set the away timeout to one minute and follow the situation with RC 7.0.0.

**Comment by Gummikavalier** [comment]
This bug happens really often but at the same it is pretty non-persistent.

Refresh always fixes it. But the status gets fixed eventually when using RC and doing other tasks meanwhile too.

I haven't been able to pinpoint it when; suddenly it just is back at green again. Possibly after the next 60 seconds timeout phase when I've been away from RC.

When I return to RC tab in browser I still see brief flash to green in RC 7.0.0 sometimes, and then it turns immediately back to Away-status in a fraction of second.

**Comment by Gummikavalier** [comment]
Our other users have noticed this too by now, so it is not just my personal use case (which often contains all kinds of fiddling with things).

**Comment by Gummikavalier** [comment]
This was fully reproducible with Chrome at https://open.rocket.chat .

The same symptoms too;
1. Away time trigger set to 61 seconds.
2. Chrome and open.rocket.chat page visible without focus.
3. After you spot status changing to Away state, hover your mouse on top of window and channel and go to that channel with mouse click.
4. Randomly instead of presence turning to green blob, it flashes very fast to green, and then sticks to orange away status.
5. No fiddling with channels and sending messages makes the presence indicator to go green at any point. (Unless you are away for a minute again and this time it may work properly.)

**Comment by Gummikavalier** [comment]
Still almost constantly happens in RC 7.1.0.

**Comment by reetp** [comment]
I'll ask Casals to look.

**Comment by imunique-ZJ** [comment]
I found a dirty workaround to fix the problem.

https://github.com/RocketChat/Rocket.Chat/blob/86a21b5784ad676907dd8f90f45a3f7aacc4a97e/apps/meteor/packages/meteor-user-presence/client/client.js#L104

If I comment out `line#104` , `line#105` and `line#106` in the `client.js` , the user status will change to `online` as expected. 

```
async 'UserPresence:online'() {
		const user = await Meteor.userAsync();
                /**
		if (user && user.status !== 'online' && user.statusDefault === 'online') {
			await Meteor.users.updateAsync({ _id: Meteor.userId() }, { $set: { status: 'online' } });
		}
                **/
		UserPresence.runCallbacks(user, 'online');
	},
```

As `'UserPresence:away'` does not call `Meteor.users.updateAsync`, maybe we can update the user status in the `ServerMethod`?

**Comment by Gummikavalier** [comment]
Still an issue in RC 7.3.1.

**Comment by reetp** [comment]
> Still an issue in RC 7.3.1.

had any response from support?

I'll flag Casals again.

**Comment by casalsgh** [comment]
taking to internal engineering team

**Comment by Gummikavalier** [comment]
Still happens in RC 7.7.0.

**Comment by reetp** [comment]
Bumped with team

**Comment by Gummikavalier** [comment]
May be early, but I _think_ this may have got fixed by RC 7.8.1. Or at least it does not happen as constantly.

**Comment by Gummikavalier** [comment]
I confirm this seems to have been fixed in RC 7.8.x. Issue closed.
